### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by dantarasov**

### CODE REVIEW

File: ui/homework/homework_2/lesson_2.md [hunk 1]
```diff
@@ -6,7 +6,11 @@
    {
         "event": "HPInteraction",
         "eventAction": "Magazines",
+<<<<<<< HEAD
```
Comment (aleksandrashmoylova) [inline ui/homework/homework_2/lesson_2.md]: here you need to fix conflicts, delete this line

File: ui/homework/homework_2/lesson_2.md [hunk 2]
```diff
@@ -6,7 +6,11 @@
...
+      npx playwright test  newsletterSubscription.test.ts  "eventCategory": "HP - D',
+=======
```
Comment (aleksandrashmoylova) [inline ui/homework/homework_2/lesson_2.md]: and this one too

File: ui/homework/homework_2/lesson_2.md [hunk 3]
```diff
@@ -6,7 +6,11 @@
...
         "eventCategory": "HP - D',
+>>>>>>> da23ca7249fdea9bea57b6f9a1f72b8e80e3b26d
```
Comment (aleksandrashmoylova) [inline ui/homework/homework_2/lesson_2.md]: and this one

File: ui/tests/createEditAcc.test.ts [hunk 4]
```diff
@@ -1,75 +1,69 @@
...
+            await page.getByPlaceholder('Email Address').fill(faker.internet.email());
+            const signUp = page.locator('//button//span//span[contains(., "Sign Up")]');
+            await signUp.click();
+        });
+
+        await test.step('create account and get event in Datalayer', async () => {
+            await page.getByPlaceholder('First name').fill(faker.person.firstName());
+            await page.getByPlaceholder('Last name').fill(faker.person.lastName());
+            await page.getByPlaceholder('Password').fill(faker.internet.password());
+            const signUp = page.locator('//button//span//span[contains(., "Sign Up")]');
```
Comment (aleksandrashmoylova) [inline ui/tests/createEditAcc.test.ts:27]: here you can use simply xpath: //button[contains(., "Sign Up")]

File: ui/tests/createEditAcc.test.ts [hunk 5]
```diff
@@ -1,75 +1,69 @@
...
+        });
+        const myAccount = page.locator('//header//button[contains(., "My Account")]');
+        await myAccount.hover();
+
+        const createButton = page.locator('//ul//a[contains(., "Create Account")]');
+        await createButton.click();
+
+        await test.step('fill email and click Sign Up', async () => {
+            await page.getByPlaceholder('Email Address').fill(faker.internet.email());
+            const signUp = page.locator('//button//span//span[contains(., "Sign Up")]');
```
Comment (aleksandrashmoylova) [inline ui/tests/createEditAcc.test.ts]: here you can use shorter xpath: //button[contains(., "Sign Up")]

File: ui/tests/createEditAcc.test.ts [hunk 6]
```diff
@@ -1,75 +1,69 @@
...
+        const [event] = await dataLayer.waitForDataLayer({
+            event: 'CategoryInteraction',
+            eventCategory: 'Login',
+            eventAction: 'Login Status',
+        });
+        expect(event).toStrictEqual(expectedEvent);
+        console.log(await page.evaluate(() => window.dataLayer));
+
+        await test.step('go to My Account', async () => {
+            const welcomeUser = page.locator('//header//button').nth(3);
```
Comment (aleksandrashmoylova) [inline ui/tests/createEditAcc.test.ts]: this xpath is not very reliable, may be you could try "//header//button[contains(., "Welcome")]" or anything else

File: ui/tests/createEditAcc.test.ts [hunk 7]
```diff
@@ -1,75 +1,69 @@
...
+            await welcomeUser.click();
+            const [window] = await page.getByLabel('Tooltip Content').all();
+            const myAccount = window.locator('[href*="/customer/account"]').first();
```
Comment (aleksandrashmoylova) [inline ui/tests/createEditAcc.test.ts]: why do use window here?

Comment (dantarasov) [reply ui/tests/createEditAcc.test.ts]: looks like it was just bad naming. Tooltip will be better

File: ui/tests/emailSubscr.test.ts [hunk 8]
```diff
@@ -37,9 +31,6 @@ test.describe('check event in Datalayer', () => {
       eventAction: 'Newsletter Subscription',
     });
     expect(event).toStrictEqual(expectedEvent);
-    /**
-     * To get all events in console
-     */
-    const a = console.log(await page.evaluate(() => window.dataLayer));
+    console.log(await page.evaluate(() => window.dataLayer));
```
Comment (LMikhail1) [inline ui/tests/emailSubscr.test.ts:34]: Unused console.log

File: ui/src/components/myDetails/myDetails.ts [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+        await this.LOCATORS.saveButton.click();
+        return newUserInfo;
+    }
+    public getFirstNameInput(): Locator {
+        return this.LOCATORS.firstNameLoc;
+    }
+    public getLastNameInput(): Locator {
+        return this.LOCATORS.lastNameLoc;
+    }
```
Comment (LMikhail1) [inline ui/src/components/myDetails/myDetails.ts:38]: You can do these methods in a different way

File: ui/src/components/myDetails/myDetails.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
+import { Component } from '@Core/component';
+import { faker } from '@faker-js/faker';
```
Comment (LMikhail1) [inline ui/src/components/myDetails/myDetails.ts:2]: Where do you import the faker?

**Pull request closed without merging**